### ISSUE TITLE
Add an option to customize editor's application region label

### ISSUE THREAD

**Issue by Comandeer** [ckeditor/ckeditor4#2445]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open `http://ckeditor.test/ckeditor-dev/samples/`.
2. Navigate into the editor.
3. Listen to JAWS announcing the editor.

### Expected result

The name of the editor is announced once.

### Actual result

The name of the editor is announced twice.

## Other details

* JAWS: 18

Iframe-based editor has its label in 3 different places:

* `[title]` attribute;
* `title` element;
* dedicated element before the editable.

Not surprising that the name's announced twice…

**Comment by Comandeer** [comment]
We decided to reorganize labels in the editor. Instead of using one label in two places, we decided to switch to two:

*   one is used as the label for the whole editor's application (editable area + toolbars),
*   one is used for the editable area only.

Each of these labels has corresponding config variable and editor's property:

| Label | Config variable | Editor's property | Default value |
| --- | --- | --- | --- |
| Application | `config.applicationTitle` | `editor.applicationTitle` | Rich Text Editor, \<editor name> |
| Editable area | `config.title` | `editor.title` | Editor, \<editor name> |

These two labels are used by assistive technology in different contexts, e.g. application label is used to differentiate between editors on page and the editable area one can be used in navigation between form controls.

Such a change also makes the screen reader output more sensible for the user – instead of reading the same label twice, now there is a difference in the two labels:

*   new approach: "Rich Text Editor, \<editor name> application. Editor, \<editor name> edit multiline",
*   old approach: "Rich Text Editor, \<editor name> application. Rich Text Editor, \<editor name> edit multiline".

Additionally, these labels are fully customizable via config variables and can be even disabled if considered redundant.

Unfortunately, this labeling system does not work well with the [`iframe`\-based editing](https://ckeditor.com/cke4/addon/wysiwygarea). For some reason, screen readers repeat the accessible name of the `iframe` for the document inside it, making the editable area label be heard twice. The best results can be achieved by using [`div`\-based editing](https://ckeditor.com/cke4/addon/divarea) – then labels are read only once.

**Comment by CKEditorBot** [comment]
Closed in https://github.com/ckeditor/ckeditor4/pull/5146
